### ISSUE TITLE
maps immediately playable after download

### ISSUE THREAD

**Issue by DanVanAtta** [triplea-game/triplea#527]
Today have to restart game or click 'refresh games' before  a map is loaded. TODO: after downloading a map, go ahead and parse it so it can be played.

**Comment by ron-murhammer** [comment]
You could also just call refresh automatically after any map is downloaded if that's easier.

**Comment by DanVanAtta** [comment]
Yip, this is a placeholder, it may require wiring of data, and it's less trivial now that the download window is non-modal.

**Comment by DanVanAtta** [comment]
Turns out maps can be played right after download. 

But, new maps will not appear in the 'choose game list' until the game model is refreshed. Refreshing the game model destroys the choose game window UI, blanks it out, and forces a person to figure out that they need to close and re-open it.

Before NewGameChooserModel.refresh() relied on strict ordering of when the game window would be open. A modal spinner blocked the window from being open while a download is in progress, and a synchronization lock blocks any new UI threads from using the game list model until it has been loaded. 

We also have code to unload this model to unload this model when the game launches, and to load it back when the game closes. This caching does not help the game launch any faster. We added in code to do a "short" parsing to support this caching feature, the game requires a "full" parsing in order to launch.

With recent updates to the game, I timed that parsing all maps takes 2.7s. Before this timing would have been very close to a minute  since a full 2GB of data would have to be read.  Now we read just the XMLs in the zip files from disk, and not the full zip file.

If we get rid of this caching, we could update the game messaging everywhere to say a map is immediately available. For now we message that the game needs to be restarted to cover the fact that the choose game list is not updated until a 'refresh game list' is clicked. If we remove the caching and load the model when the choose game window is opened, then the new map will become available in that list without having to click 'refresh maps'.

Also for consideration, we spawn whole new java processes when someone joins the lobby, and again when they join/host a game. That is 2 more copies of the full set of map data, and also code that controls whether that data should be loaded/flushed. We actually unload it explicitly for memory, yet we have duplicates of the data.

~3s is pretty slow, though the cache I don't think is as necessary or as useful as it once was. When the load time was typically around 10 seconds for just a few maps, I could see us needing it for sure, now though, it's much more debatable. I think we are probably better removing the cache, working things down, then looking to see if we could improve that 3s load time further.
